apache-http-client/src/main/java/com/joyent/http/signature/apache/httpclient/HttpSignatureAuthScheme.java

```diff
@@ -54,6 +68,13 @@
     private final ThreadLocalSigner signer;
 
     /**
+     * Map of credentials to cache object used for looking up cached signatures.
+     */
+    private ConcurrentMap<Credentials, HttpSignatureCache> signatureCacheMap =
```

Comment:
Given the lack of good options (more 3rd party deps), waiting for real use cases sounds okay to me.